pkg/reconciler/knativeserving/knativeserving.go

```diff
@@ -109,7 +109,11 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ks *v1alpha1.KnativeServ
 		common.CheckDeployments,
 		common.DeleteObsoleteResources(ctx, ks, r.installed),
 	}
-	manifest := r.manifest.Append()
+	platformManifests, err := r.extension.Manifests(ks)
```

Comment:
@skonto plz update your local source code and do a rebase.
Your code is surely obsolete for now.

Another remark: as in the stages, we load the core manifests first, then the ingress manifests(you should have them after the rebase). And probably the extended manifests. It is order-critical. The one loaded first will be applied first, the one loaded in the end will overwrite the same ones before.
The question is if we want to make sure the extended manifest go over the core, we need to load them in the order: core first, ingress manifest second, and lastly the extended manifest.
But for the logic in this PR, it seems to be extended manifest first, core second... Considering changing the order.
You can create a function similar to `common.AppendTarget` for extension, and add it in the stage.

Reply:
```
func (r *Reconciler) appendExtension(ctx context.Context, manifest *mf.Manifest, instance v1alpha1.KComponent) error {
	platformManifests, err := r.extension.Manifests(instance)
	if err != nil {
		return err
	}
	*manifest = manifest.Append(platformManifests)
	return nil
}
```

```
	stages := common.Stages{
		common.AppendTarget,
		ingress.AppendTargetIngresses,
		r.filterDisabledIngresses,
                r.appendExtension,
		r.transform,
		common.Install,
		common.CheckDeployments,
		common.DeleteObsoleteResources(ctx, ks, r.installed),
	}
```

Reply:
thanks @houshengbo will take a look.